### PR TITLE
Fix bug in deleting clusters in unowned_cluster integration test

### PR DESCRIPTION
**Pull request by aclevername**
### Description
@Callisto13 :thinking: 

### Checklist
- [ ] Added tests that cover your change (if possible)
- [ ] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [ ] Manually tested
- [ ] Made sure the title of the PR is a good description that can go into the release notes
- [ ] (Core team) Added labels for change area (e.g. `area/nodegroup`) and kind (e.g. `kind/improvement`)

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

### CODE REVIEW

**Comment by Callisto13** [comment]
All part of the plan

**Review by Callisto13 — DISMISSED** [review]

File: integration/tests/unowned_cluster/unowned_cluster_test.go [hunk 1]
```diff
@@ -66,7 +66,7 @@ var _ = Describe("(Integration) [non-eksctl cluster & nodegroup support]", func(
 	})
 
 	AfterSuite(func() {
-		if !params.SkipDelete {
```
Comment (michaelbeaumont) [inline integration/tests/unowned_cluster/unowned_cluster_test.go:69]: wait why is this correct?

File: integration/tests/unowned_cluster/unowned_cluster_test.go [hunk 2]
```diff
@@ -287,7 +287,7 @@ var _ = Describe("(Integration) [non-eksctl cluster & nodegroup support]", func(
 	})
 
 	It("supports deleting clusters", func() {
-		if !params.SkipDelete {
```
Comment (michaelbeaumont) [inline integration/tests/unowned_cluster/unowned_cluster_test.go:290]: :100:

File: integration/tests/unowned_cluster/unowned_cluster_test.go (hunk 1 above)
Comment (Callisto13) [reply integration/tests/unowned_cluster/unowned_cluster_test.go:69]: wait actually yeh. this is fine jake i am not completely rubbish

Comment (aclevername) [reply integration/tests/unowned_cluster/unowned_cluster_test.go:69]: The stack only gets created if `SkipCreate` is true, so we should only delete is if `SkipCreate` is true

Comment (Callisto13) [reply integration/tests/unowned_cluster/unowned_cluster_test.go:69]: but what if you wanted it created, but not deleted?

Comment (aclevername) [reply integration/tests/unowned_cluster/unowned_cluster_test.go:69]: then we should switch on both? :see_no_evil:

Comment (Callisto13) [reply integration/tests/unowned_cluster/unowned_cluster_test.go:69]: > The stack only gets created if SkipCreate is true
when it's `false` probably

i think skipping delete of things when `SkipDelete` is `true` is the way things should work

File: integration/tests/unowned_cluster/unowned_cluster_test.go (hunk 2 above)
Comment (Callisto13) [reply integration/tests/unowned_cluster/unowned_cluster_test.go:290]: actually this is right too

@aclevername i suggest that this PR has no merit!

Comment (Callisto13) [reply integration/tests/unowned_cluster/unowned_cluster_test.go:290]: this should check for `true`, so it is doing the right thing

File: integration/tests/unowned_cluster/unowned_cluster_test.go (hunk 1 above)
Comment (michaelbeaumont) [reply integration/tests/unowned_cluster/unowned_cluster_test.go:69]: `!Skip` is already way too many negations to keep together in my head

File: integration/tests/unowned_cluster/unowned_cluster_test.go (hunk 2 above)
Comment (michaelbeaumont) [reply integration/tests/unowned_cluster/unowned_cluster_test.go:290]: Shouldn't it just be `SkipDelete || SkipCreate` like if I skip create then I probably want to skip delete? (I'm struggling to imagine when I want to `SkipCreate` in the first place tbh...)

Comment (Callisto13) [reply integration/tests/unowned_cluster/unowned_cluster_test.go:290]: oh yeh actually nvm

Comment (Callisto13) [reply integration/tests/unowned_cluster/unowned_cluster_test.go:290]: i read it wrong

Comment (aclevername) [reply integration/tests/unowned_cluster/unowned_cluster_test.go:290]: I don't get the use case for the various Skips. But it makes sense to only skip deleting the *cluster* if `SkipDelete` is true

File: integration/tests/unowned_cluster/unowned_cluster_test.go (hunk 1 above)
Comment (aclevername) [reply integration/tests/unowned_cluster/unowned_cluster_test.go:69]: MY BRAIN IS MELTING:

We should only delete the the stack (*different to the cluster*) if:
1. It exists, so if SkipCreate was True it doesn't exist, so we shouldn't delete it
2. SkipDelete is true

File: integration/tests/unowned_cluster/unowned_cluster_test.go [hunk 3]
```diff
@@ -66,7 +66,7 @@ var _ = Describe("(Integration) [non-eksctl cluster & nodegroup support]", func(
 	})
 
 	AfterSuite(func() {
-		if !params.SkipDelete {
+		if !params.SkipCreate || !params.SkipDelete {
```
Comment (michaelbeaumont) [inline integration/tests/unowned_cluster/unowned_cluster_test.go:69]: | SkipCreate       | SkipDelete           | deleteStack  |
| ------------- |:-------------:| -----:|
| false      | false | true |
| false     | true      |   false |
| true |  false   |    false/true? | 
| true |  true   |    false |

Comment (aclevername) [reply integration/tests/unowned_cluster/unowned_cluster_test.go:69]: We don't have a stack to delete if SkipCreate is true, so the 3rd row outcome is `false`

Comment (michaelbeaumont) [reply integration/tests/unowned_cluster/unowned_cluster_test.go:69]: ok so then it should be `!params.SkipCreate && !params.SkipDelete` :dart:

File: integration/tests/unowned_cluster/unowned_cluster_test.go [hunk 4]
```diff
@@ -287,7 +287,7 @@ var _ = Describe("(Integration) [non-eksctl cluster & nodegroup support]", func(
 	})
 
 	It("supports deleting clusters", func() {
-		if !params.SkipDelete {
+		if params.SkipDelete {
```
Comment (michaelbeaumont) [inline integration/tests/unowned_cluster/unowned_cluster_test.go:290]: | SkipCreate       | SkipDelete           | skip this test  |
| ------------- |:-------------:| -----:|
| false      | false | false |
| false     | true      |   true |
| true |  false   |  false/true?   | 
| true |  true   |  true  |

If I have `SkipCreate` do I have a cluster to delete?

File: integration/tests/unowned_cluster/unowned_cluster_test.go (hunk 3 above)
Comment (aclevername) [reply integration/tests/unowned_cluster/unowned_cluster_test.go]: oops :+1:

File: integration/tests/unowned_cluster/unowned_cluster_test.go (hunk 4 above)
Comment (aclevername) [reply integration/tests/unowned_cluster/unowned_cluster_test.go:290]: Yes because you provided a cluster

Comment (michaelbeaumont) [reply integration/tests/unowned_cluster/unowned_cluster_test.go:290]: i'd say we don't delete it then? since it's "managed" outside of the tests?

Comment (aclevername) [reply integration/tests/unowned_cluster/unowned_cluster_test.go:290]: but then `SkipCreate` is really `SkipCreateAndDelete`

Comment (michaelbeaumont) [reply integration/tests/unowned_cluster/unowned_cluster_test.go:290]: yeah basically or seen another way, the default of `SkipDelete` actually _depends_ on the value of `SkipCreate`

Comment (aclevername) [reply integration/tests/unowned_cluster/unowned_cluster_test.go:290]: the only other test using this functionality is `crud` and that follows the `SkipCreate` only means skip creating, not deleting too. I think this is correct for now.

Really this whole thing should get refactored, but thats a different PR

Comment (michaelbeaumont) [reply integration/tests/unowned_cluster/unowned_cluster_test.go:290]: ok!

**Review by michaelbeaumont — APPROVED** [review]
seems about right!

**Comment by aclevername** [comment]
> seems about right!

I'll take it! Merging.

**Pull request merged**